### PR TITLE
remove weight parallelism

### PR DESCRIPTION
**Pull request by eitanturok**
# What does this PR do?

This PR removes weight parallelism as we never use it. Tagging @tgale96.

Since we use FSDP's weight parallelism and not our own custom weight parallelism in MegaBlocks, I wanted to remove the weight parallelism implementation.

Specifically, we
1. Remove `test_parallelism.py` because this file tests that weight parallelism and expert parallelism have the same results.
2. Remove `moe_weight_parallelism` and `weight_parallel_group` from the args.
3. Remove weight parallelism from all the layers.

Because `moe_weight_parallelism` is `False` by default and `weight_parallel_group` is None by default, 
`mpu.get_weight_parallel_world_size(args)` always returned 1 and `mpu.get_weight_parallel_rank(args)` always returns 0. This allowed us to drastically simplify things in `mlp.create_dmoe_expert_weights()`.

Also, can I get an extra close review of my changes to the `MemoryOptimizedMLP.parallel_forward()` method? I noticed that the group would always be `None` there but I am hesitant to hard-code this in. Not sure if this is the right thing to do.

Also, I ran all tests locally and they pass.

(Also, enjoy this nice PR template that I added!)

# What issue(s) does this change relate to?

# Before submitting
- [x] Have you read the [contributor guidelines](https://github.com/databricks/megablocks/blob/dev/CONTRIBUTING.md)?
- [ ] Is this change a documentation change or typo fix? If so, skip the rest of this checklist.
- [ ] Was this change discussed/approved in a GitHub issue first? It is much more likely to be merged if so.
- [ ] Did you update any related docs and document your change?
- [x] Did you update any related tests and add any new tests related to your change? (see [testing](https://github.com/databricks/megablocks/blob/dev/CONTRIBUTING.md#running-tests))
- [x] Did you run the tests locally to make sure they pass?
- [x] Did you run `pre-commit` on your change? (see the `pre-commit` section of [prerequisites](https://github.com/databricks/megablocks/blob/dev/CONTRIBUTING.md#prerequisites))

### CODE REVIEW

**Comment by tgale96** [comment]
So this looks great but I think you can remove a lot more code!

[weight_parallel.py](https://github.com/databricks/megablocks/blob/main/megablocks/layers/weight_parallel.py) can be completely removed, I think. And then all uses of it, including the `parallel_forward` function you called out, which I don't think is used anywhere anyways?

Deleting that file will help you track down every use of weight parallelism as well, since it all gets routed into that one.

**Comment by eitanturok** [comment]
* removed parallel_forward
* deleted weight_parallel.py
All tests pass!

**Comment by tgale96** [comment]
LGTM! The last thing I might do is to just `grep weight_parallel`, if you haven't already. But I think you got everything.

**Comment by eitanturok** [comment]
Already grep-ed!

File: megablocks/layers/mlp.py [hunk 1]
```diff
@@ -181,20 +180,7 @@ def create_dmoe_expert_weights(
         init_method,
     )
     weights = weights.view([-1, columns])
```
[the remark this answers is not in the document]
Comment (eitanturok) [reply megablocks/layers/mlp.py]: Fixed.

**Review by mvpatel2000 — APPROVED** [review]
LGTM

**Review by mihir-db — APPROVED** [review]

**Pull request merged**